File: matiasmasso/Mat.NET/DT2/DTOSpvIn.vb
Public Class DTOSpvIn
    Inherits DTOBaseGuid

    'Public Shadows Property Guid As Guid
    'Public Shadows Property Guid As Guid

    Property Emp As DTOEmp
    Property Id As Integer
    Property Fch As Date
    Property Expedicio As String
    Property Bultos As Integer
    Property Kg As Integer
    Property M3 As Decimal
    Property Obs As String
    Property Spvs As List(Of DTOSpv)

    Property SpvCount As Integer
    Property UsrNum As Integer
    Property User As DTOUser

    Shared Function Factory() As DTOSpvIn
        Dim retval As New DTOSpvIn
        retval.Guid = Guid.NewGuid
        Return retval
    End Function

    Public Sub New()
        MyBase.New()
        _Spvs = New List(Of DTOSpv)
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
        _Spvs = New List(Of DTOSpv)
    End Sub

    Shared Function Factory(oUser As DTOUser, DtFch As Date) As DTOSpvIn
        Dim retval As New DTOSpvIn
        With retval
            .User = oUser
            .Emp = oUser.Emp
            .Fch = DtFch
        End With
        Return retval
    End Function

    Shared Function DeleteQuery(src As List(Of DTOSpvIn)) As String
        Dim retval As String = ""
        Select Case src.Count
            Case 0
                retval = "No hi han entrades sel·leccionades"
            Case 1
                retval = String.Format("retrocedim la entrada {0}?", src.First.Id)
            Case Else
                retval = "retrocedim les entrades sel·leccionades?"
        End Select
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOSpvIn.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace DTO
{
    public class DTOSpvIn : DTOBaseGuid
    {

        // Public Shadows Property Guid As Guid
        // Public Shadows Property Guid As Guid

        public DTOEmp emp { get; set; }
        public int id { get; set; }
        public DateTime fch { get; set; }
        public string expedicio { get; set; }
        public int bultos { get; set; }
        public int kg { get; set; }
        public decimal m3 { get; set; }
        public string obs { get; set; }
        public List<DTOSpv> spvs { get; set; }

        public int spvCount { get; set; }
        public int usrNum { get; set; }
        public DTOUser user { get; set; }

        public static DTOSpvIn Factory()
        {
            DTOSpvIn retval = new DTOSpvIn();
            retval.Guid = Guid.NewGuid();
            return retval;
        }

        public DTOSpvIn() : base()
        {
            spvs = new List<DTOSpv>();
        }

        public DTOSpvIn(Guid oGuid) : base(oGuid)
        {
            spvs = new List<DTOSpv>();
        }

        public static DTOSpvIn Factory(DTOUser oUser, DateTime DtFch)
        {
            DTOSpvIn retval = new DTOSpvIn();
            {
                var withBlock = retval;
                withBlock.user = oUser;
                withBlock.emp = oUser.Emp;
                withBlock.fch = DtFch;
            }
            return retval;
        }

        public static string deleteQuery(List<DTOSpvIn> src)
        {
            string retval = "";
            switch (src.Count)
            {
                case 0:
                    {
                        retval = "No hi han entrades sel·leccionades";
                        break;
                    }

                case 1:
                    {
                        retval = string.Format("retrocedim la entrada {0}?", src.First().id);
                        break;
                    }

                default:
                    {
                        retval = "retrocedim les entrades sel·leccionades?";
                        break;
                    }
            }
            return retval;
        }
    }
}
